From fd8c89b6080a9c1a58b4827f99c825df0bc54927 Mon Sep 17 00:00:00 2001
From: Pooventhiran G <quic_pooventh@quicinc.com>
Date: Thu, 8 Aug 2024 12:15:03 +0530
Subject: [PATCH] cfg80211: Enable Single Wiphy MLO Reconfig APIs

Enable Single Wiphy MLO Reconfig APIs in rdev-ops and cfg80211 for
offloaded drivers to support MLO reconfig.

Change-Id: I1d038d2b6e2a095e9ebfde796a82c42683da58fc
Signed-off-by: Pooventhiran G <quic_pooventh@quicinc.com>
---
 include/net/cfg80211.h | 4 ++++
 net/wireless/Makefile  | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/include/net/cfg80211.h b/include/net/cfg80211.h
index c1e21dc3dcb0..a12a7780060c 100644
--- a/include/net/cfg80211.h
+++ b/include/net/cfg80211.h
@@ -25,6 +25,10 @@
 #include <linux/rfkill.h>
 #include <net/regulatory.h>
 
+#ifndef CONFIG_ML_RECONFIG_SINGLE_WIPHY
+#define CONFIG_ML_RECONFIG_SINGLE_WIPHY 1
+#endif
+
 /**
  * DOC: Introduction
  *
diff --git a/net/wireless/Makefile b/net/wireless/Makefile
index 527ae669f6f7..ad5f4d3d46b9 100644
--- a/net/wireless/Makefile
+++ b/net/wireless/Makefile
@@ -24,6 +24,8 @@ ifneq ($(CONFIG_CFG80211_EXTRA_REGDB_KEYDIR),)
 cfg80211-y += extra-certs.o
 endif
 
+CFLAGS += -DCONFIG_ML_RECONFIG_SINGLE_WIPHY=1
+
 $(obj)/shipped-certs.c: $(wildcard $(srctree)/$(src)/certs/*.hex)
 	@$(kecho) "  GEN     $@"
 	$(Q)(echo '#include "reg.h"'; \
-- 
2.34.1

